src/shared/clock-util.c

```diff
@@ -139,10 +139,12 @@ int clock_reset_timewarp(void) {
 
 #define EPOCH_FILE "/usr/lib/clock-epoch"
 
-int clock_apply_epoch(void) {
+int clock_apply_epoch(ClockChangeDirection *ret_attempted_change) {
         struct stat st;
         struct timespec ts;
-        usec_t epoch_usec;
+        usec_t epoch_usec, now_usec, diff_usec;
+
```

Comment:
so usually we have the rule that we do not mangle return params on failures. In this case that's kinda impotant though. Hence please add an explicit comment here, i.e. add something like this here:

```c
/* NB: we update *ret_attempted_change in *all* cases, both on success and failure, to indicate what we intended to do! */
```